### PR TITLE
restore the lock_thread value after rollback

### PR DESCRIPTION
**Pull request by cou929**
Fixes #232

### What is the purpose of this pull request?

Enables testing of database connections from multiple threads in test cases that do not use before_all.

### What changes did you make? (overview)

Restore the value of `ActiveRecord::Base.connection.pool.lock_thread` after rollback.

### Checklist

- [x] I've added tests for this change
- [x] I've added a Changelog entry
- [ ] I've updated a documentation

### CODE REVIEW

File: lib/test_prof/before_all/adapters/active_record.rb [hunk 1]
```diff
@@ -44,8 +44,14 @@ def setup_fixtures(test_object)
       # might lead to leaking connections
       config.before(:begin) do
         next unless ::ActiveRecord::Base.connection.pool.respond_to?(:lock_thread=)
+        @orig_lock_thread = ::ActiveRecord::Base.connection.pool.instance_variable_get("@lock_thread")
```
Comment (palkan) [inline lib/test_prof/before_all/adapters/active_record.rb:47]: Can we be a bit more defensive here and use a more unique name for the instance variable? (Or even some esoteric one like [here](https://github.com/test-prof/test-prof/blob/e47c9ee3de9d3a2dad0822bb4c9bd93259938210/lib/test_prof/recipes/rspec/let_it_be.rb#L78)?

**Review by palkan — APPROVED** [review]
LGTM 👍 Thanks!

Let's add a changelog entry (and make RuboCop happy 🙂)

File: lib/test_prof/before_all/adapters/active_record.rb (hunk 1 above)
Comment (cou929) [reply lib/test_prof/before_all/adapters/active_record.rb]: > Or even some esoteric one like [here](https://github.com/test-prof/test-prof/blob/e47c9ee3de9d3a2dad0822bb4c9bd93259938210/lib/test_prof/recipes/rspec/let_it_be.rb#L78)?

wow, that's great 😸

**Comment by cou929** [comment]
- [3d54ebc](https://github.com/test-prof/test-prof/pull/236/commits/3d54ebc80a5ee243a1a9c67cbfce01459ae489ab)
    - fix `[Correctable] Performance/StringIdentifierArgument: Use :@lock_thread instead of "@lock_thread".` (ref. https://github.com/test-prof/test-prof/runs/5221166673?check_suite_focus=true)
- [616bbb0](https://github.com/test-prof/test-prof/pull/236/commits/616bbb0988bea6b099406c3a6e0702e5e8052468)
    - fix instance variable name (ref. https://github.com/test-prof/test-prof/pull/236#discussion_r808338985)
- [14afbcc](https://github.com/test-prof/test-prof/pull/236/commits/14afbccc5cde7929a46e7dfb77b1f719fb795e20)
    - added a Changelog entry

**Comment by palkan** [comment]
Thanks!

**Pull request merged**